File: Giới thiệu/prime.cpp
// Thuật toán tối ưu nhất
#include <iostream>
#include <cmath>

using namespace std;

bool prime(long long int number) {
  if(number < 2) return false;
  for(int i = 2; i <= sqrt(number); i++){
    if(number % i == 0) return false;
  }
  return true;
}

int main(){
  long long int number;
  cin >> number;
  if(prime(number)) {
    cout << "True";
  }
  else {
    cout << "False";
  }
  return 0;
}

File: introduce/prime.java
package introduce;

import java.util.Scanner;

class prime {
    boolean isPrime(long number) {
        if (number < 2) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(number); i++) {
            if (number % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        long number = input.nextLong();
        prime obj = new prime();
        boolean result = obj.isPrime(number);
        if (result) {
            System.out.println("True");
        }
        else {
            System.out.println("False");
        }

        input.close();
    }
}